lambda/fulfillment/lib/middleware/lex.js

```diff
@@ -191,6 +214,57 @@ function buildImageResponseCardV2(response) {
     return imageResponseCardV2;
 }
 
+function buildInteractiveMessageElements(elements){
+    return elements.map(x => ({title: x.text}));
+}
+
+function buildInteractiveMessageTemplate(response){
+    response = limitInteractiveMessagesDisplayTextLength(response);
+
+    if(response.message.length > 100){
+        qnabot.log("WARNING: Truncating message content to Interactive Message Title limit of 100 characters");
+    }
+
+    let template = {
+        templateType: "ListPicker",
+        version: "1.0",
+        data: {
+            content: {
+                title: response.message.slice(0,99),
+                elements: buildInteractiveMessageElements(_.get(response.card,"buttons")),
+            },
+        },
+    };
+    if(_.get(response,"card.title",undefined)!== undefined){
+        if( _.get(response,"card.title").length > 200){
+            qnabot.log("WARNING: Truncating Card Title to Interactive Message Subtitle limit of 200 characters");
+        }
+        template.data.content.subtitle = _.get(response,"card.title").slice(0,199);
+    }
+    if(_.get(response,"card.imageUrl",undefined)!== undefined){
+        template.data.content.imageType = "URL";
+        template.data.content.imageData = _.get(response,"card.imageUrl");
+    }
+    
+    return JSON.stringify(template);
+}
+
+function buildV1InteractiveMessageResponse(request, response) {
```

Comment:
Are the functions for V1 and V2 the same and do we expect them to be the same in the future? If so, is there a need for two functions?

Reply:
Yes.  The expected format for V1 vs V2 is different.